src/dmd/lexer.d

```diff
@@ -423,7 +423,26 @@ class Lexer
                     p += 3;
                 }
                 else
+                {
                     t.value = charConstant(t);
+                    if (t.unsvalue == '?' && errors())
+                    {
+                        while (*p != 0x1A && *p != 0
```

Comment:
The failure might be because you need to increment the line number here somewhere.

Reply:
Yes, I think you are right. I'll stop the search when I'll encounter a newline and I'll handle this case separately to increment the line number.